### PR TITLE
Feature/fix screenshot

### PR DESCRIPTION
**Pull request by enesoezel**

### CODE REVIEW

File: src/main/java/de/retest/recheck/ui/descriptors/AttributeUtil.java [hunk 1]
```diff
@@ -2,8 +2,34 @@
...
+				(IdentifyingAttributesDifference) difference.getIdentifyingAttributesDifference();
+		for ( final AttributeDifference aDiff : diff.getAttributeDifferences() ) {
+			if ( aDiff.getKey().equals( OutlineAttribute.RELATIVE_OUTLINE ) ) {
+				return ((Rectangle) aDiff.getActual());
+			}
+		}
+		return getOutline( difference.getIdentifyingAttributes() );
+	}
+
+	public static Rectangle getActualAbsoluteOutline( final ElementDifference difference ) {
```
Comment (beatngu13) [inline src/main/java/de/retest/recheck/ui/descriptors/AttributeUtil.java:21]: `getActualOutline` and this method are quite similar. Does it make sense to merge/reuse parts?

File: src/main/java/de/retest/recheck/ui/diff/ElementDifference.java [hunk 2]
```diff
@@ -56,9 +56,9 @@ public Screenshot mark( final Screenshot screenshot, final Filter filter ) {
 		if ( childDifferences != null ) {
 			for ( final Difference childDifference : childDifferences ) {
 				for ( final ElementDifference compDiff : childDifference.getNonEmptyDifferences() ) {
-					if ( !filter.matches( element )
```
Comment (beatngu13) [inline src/main/java/de/retest/recheck/ui/diff/ElementDifference.java:59]: I think this diff doesn't belong in this commit?

File: src/main/java/de/retest/recheck/ui/descriptors/Attribute.java [hunk 3]
```diff
@@ -10,7 +10,7 @@
 
 	private static final long serialVersionUID = 1L;
 
-	public static final double IGNORE_WEIGHT = 0.0d;
+	public static final double IGNORE_WEIGHT = Float.MIN_NORMAL;
```
Comment (beatngu13) [inline src/main/java/de/retest/recheck/ui/descriptors/Attribute.java]: Could you explain this change, commit message isn't very helpful? And isn't there also a `Double` variant?

**Review by beatngu13 — CHANGES_REQUESTED** [review]

File: src/main/java/de/retest/recheck/ui/descriptors/AttributeUtil.java [hunk 4]
```diff
@@ -9,24 +10,32 @@
 public class AttributeUtil {
 
 	public static Rectangle getActualOutline( final ElementDifference difference ) {
-		final IdentifyingAttributesDifference diff =
+		final IdentifyingAttributesDifference identifyingAttributesDifference =
 				(IdentifyingAttributesDifference) difference.getIdentifyingAttributesDifference();
-		for ( final AttributeDifference aDiff : diff.getAttributeDifferences() ) {
-			if ( aDiff.getKey().equals( OutlineAttribute.RELATIVE_OUTLINE ) ) {
-				return ((Rectangle) aDiff.getActual());
+		try {
```
Comment (diba1013) [inline src/main/java/de/retest/recheck/ui/descriptors/AttributeUtil.java]: Why is there a `try catch`?

File: src/main/java/de/retest/recheck/ui/descriptors/AttributeUtil.java [hunk 5]
```diff
@@ -1,9 +1,44 @@
...
+import de.retest.recheck.ui.diff.ElementDifference;
+import de.retest.recheck.ui.diff.IdentifyingAttributesDifference;
 
 public class AttributeUtil {
 
+	public static Rectangle getActualOutline( final ElementDifference difference ) {
+		final IdentifyingAttributesDifference identifyingAttributesDifference =
+				(IdentifyingAttributesDifference) difference.getIdentifyingAttributesDifference();
+		try {
+			if ( Objects.nonNull( identifyingAttributesDifference.getAttributeDifferences() ) ) {
```
Comment (diba1013) [inline src/main/java/de/retest/recheck/ui/descriptors/AttributeUtil.java]: Just use `!= null` since there is no advantage to use `nonNull`. Also extract `identifyingAttributesDifference.getAttributeDifferences()` since you call it twice.

**Review by diba1013 — CHANGES_REQUESTED** [review]

**Comment by beatngu13** [comment]
Is this PR related to #195, which has been merged recently?
